### PR TITLE
Enable noUncheckedIndexedAccess for Local Driver

### PR DESCRIPTION
**Pull request by RishhiB**
Enable noUncheckedIndexedAccess for Local Driver
We are enabling `noUncheckedIndexedAccess` to protect us from changing a type of a record where there was a named property and reduce the risk of runtime errors by ensuring that indexed access on objects are properly checked for undefined values.

Arrays are impacted by this and must be annotated even though it is not the reason for using the setting

[AB#8216](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8216)

### CODE REVIEW

File: packages/drivers/local-driver/src/localCreateDocument.ts [hunk 1]
```diff
@@ -20,8 +20,12 @@ export async function createDocument(
 ) {
 	const pathName = new URL(resolvedUrl.url).pathname;
 	const pathArr = pathName.split("/");
-	const tenantId = pathArr[pathArr.length - 2];
-	const id = pathArr[pathArr.length - 1];
+	// Why are we non null asserting here
+	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	const tenantId = pathArr[pathArr.length - 2]!;
+	// Why are we non null asserting here
```
Comment (pragya91) [inline packages/drivers/local-driver/src/localCreateDocument.ts:26]: Why is this text here?

File: packages/drivers/local-driver/src/localCreateDocument.ts [hunk 2]
```diff
@@ -20,8 +20,12 @@ export async function createDocument(
 ) {
 	const pathName = new URL(resolvedUrl.url).pathname;
 	const pathArr = pathName.split("/");
-	const tenantId = pathArr[pathArr.length - 2];
-	const id = pathArr[pathArr.length - 1];
+	// Why are we non null asserting here
```
Comment (pragya91) [inline packages/drivers/local-driver/src/localCreateDocument.ts:23]: Why is this text here?

File: packages/drivers/local-driver/src/localCreateDocument.ts (hunk 1 above)
Comment (RishhiB) [reply packages/drivers/local-driver/src/localCreateDocument.ts:26]: See https://github.com/microsoft/FluidFramework/pull/21659#discussion_r1664495176

File: packages/drivers/local-driver/src/localCreateDocument.ts (hunk 2 above)
Comment (RishhiB) [reply packages/drivers/local-driver/src/localCreateDocument.ts]: Please see https://github.com/microsoft/FluidFramework/pull/21659#discussion_r1664495176

File: packages/drivers/local-driver/src/localDocumentStorageService.ts [hunk 3]
```diff
@@ -91,12 +93,16 @@ export class LocalDocumentStorageService implements IDocumentStorageService {
 				throw new Error("No versions for the document!");
 			}
 
-			versionId = versions[0].treeId;
+			// Non null asserting here because of the length check above
```
Comment (scottn12) [inline packages/drivers/local-driver/src/localDocumentStorageService.ts:96]: We should use the standard comments

**Review by scottn12 — APPROVED** [review]

File: packages/drivers/local-driver/src/localDocumentStorageService.ts (hunk 3 above)
Comment (RishhiB) [reply packages/drivers/local-driver/src/localDocumentStorageService.ts:96]: This will be the final comment so I am not adding a TODO

**Pull request merged**